app/response_processor.py

```diff
@@ -21,9 +21,13 @@ def options():
             pass
         return rv
 
-    def __init__(self, logger):
+    def __init__(self, logger, tx_id=None):
         self.logger = logger
-        self.tx_id = ""
+
+        if tx_id:
+            self.tx_id = tx_id
+        else:
+            self.tx_id = None
```

Comment:
Expected behavior is that collect now gets tx_id from message props.

If for any reason this isn't defined, it will then try to determine tx_id from the document itself.

Relevant code is:
```python
if not self.tx_id:
    self.tx_id = decrypted_json.get('tx_id')
elif self.tx_id != decrypted_json.get('tx_id'):
    raise BadMessageError
else:
    self.logger = self.logger.bind(tx_id=self.tx_id)

Code update done
```